src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,139 @@
+
+        }
+        return categoryOB.success_add();
+    }
+    /**
+     * Overrides method in Category_IB.
+     * Attempts to edit a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
+     *  1. Repeated Name: User tries to edit category name to another name that exists in the month.
+     *  2. NoSuchElementException: User tries to edit a category that does not exist.
+     *  3. NumberFormatException: User tries to edit a budget value with input that can not be converted to a double.
+     *  4. Category budget less than 0: User tries to edit a budget value with input that is a negative number.
+     *
+     * @param categoryID_edit Category_Input_Data required for editing a new category to the designated monthID MonthlyStorage Object.
+     * @return Category_Output_Data category.
+     */
+    @Override
+    public CategoryOD editCategoryInMonth(CategoryID categoryID_edit) throws EntityException {
+        MonthlyStorage month = categoryID_edit.getSession().getMonthlyData(categoryID_edit.getMonthID());
+        ArrayList<Category> category_list = month.getCategoryData();
+        try {
+            for (Category e : category_list) {
+                if (e.getName().equals(categoryID_edit.getName())) {
+                    //1. Repeated name fail
+                    categoryOB.fail("There is already a category with this new name in this month.");
+                }
+            }
+
+            Category c = findCategory(category_list, categoryID_edit.getOld_category().getName());
+
+            double value_double = toDouble(categoryID_edit.getValue());
+            if (value_double < 0) {
+                //4. Category budget less than 0 fail
+                categoryOB.fail("Category budget can't be less than $0. Please try again!");
+            }
+
+            c.setName(categoryID_edit.getName());
+
+            categoryID_edit.setValue(value_double);
+            c.setBudget(value_double);
+
+        } catch (NoSuchElementException e) {
+            //2. NoSuchElementException fail
+            categoryOB.fail("There is no such category in the current month. Please add a new category or select existing category!");
+        } catch(NumberFormatException e){
+            //3. NumberFormatException fail
+            categoryOB.fail("Category budget needs to be a number. Please try again!");
+
+        }
+        return categoryOB.success_edit();
+    }
+
+    /**
+     * Helper method returns a Category from a given category list of a MonthlyStorage Object.
+     * @param monthCategoryData An ArrayList of categories.
+     * @param name Category name.
+     * @return Category with given String name.
+     */
+    @Override
+    public Category findCategory(ArrayList<Category> monthCategoryData, String name){
```

Comment:
Nice helper method!

Reply:
Oh, I just noticed, but it's missing the throws NoSuchElementException in the function header